### PR TITLE
Prevent off-by-one error on comments on newly appended lines

### LINKED ISSUE #17875: pull requests: wrong line numbers in discussions
### Gitea Version

1.16.0+dev-630-g042cac5fe

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

tested on try.gitea.io with a migrated repo

### Database

_No response_

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Description

demo: https://try.gitea.io/murks/Vereinsdokumente/pulls/7

I've created a comment for line 23 in a PR. While everything is fine in the "Files Changed" tab, the text block in the conversation view is off-by-one which brings the comments out of context.

### Screenshots

`Files Changed` tab:
image

`Conversation` tab with wrong text section:
image

### PR DESCRIPTION
**Pull request by zeripath**
There was a bug in CutDiffAroundLine whereby if a file without a terminal new line
has a patch which appends lines to it and a comment is placed on one of those lines
the comment diff will be a line out of place.

This fixes CutDiffAroundLine to simply ignore the missing terminal newline - however,
we should really improve this rendering to add a marker to say that there was a
previously missing terminal newline.

Fix #17875

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: modules/git/diff.go [hunk 1]
```diff
@@ -152,6 +152,7 @@ func CutDiffAroundLine(originalDiff io.Reader, line int64, old bool, numbersOfLi
 
 	for scanner.Scan() {
 		lof := scanner.Text()
+		fmt.Printf("%03d: %s\n", currentLine, lof)
```
Comment (KN4CK3R) [inline modules/git/diff.go]: Debug?

Comment (zeripath) [reply modules/git/diff.go]:
```suggestion
```

File: modules/git/diff_test.go [hunk 2]
```diff
@@ -42,6 +42,58 @@ index d8e4c92..19dc8ad 100644
...
++
++	3. Online-Veranstaltung: Keine, parallele Online-Veranstaltung, ausschließlich Online-Veranstaltung
++
++5. Bei Präsenzveranstungen gelten außerdem die Hygienevorschriften des Veranstaltungsorts. Bei Regelkollision greift die restriktivere Regel.
+\ No newline at end of file`
+
+func TestCutDiffAroundLineIssue17875(t *testing.T) {
+	result, err := CutDiffAroundLine(strings.NewReader(issue17875Diff), 23, false, 3)
+	assert.NoError(t, err)
+	// resultByLine := strings.Split(result, "\n")
```
Comment (KN4CK3R) [inline modules/git/diff_test.go]:
```suggestion
```

**Review by KN4CK3R — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
Make lgtm work

**Pull request merged**